### ISSUE TITLE
[FEAT] Add model module

### ISSUE THREAD

**Issue by fabiofelix** [VIDA-NYU/UrbanMapper#76]
# 💡 Feature Request

**Welcome to the `UrbanMapper` feature request hub!** 🎉

Hey there, before you jump in, take a quick sec to check out the
important stuff below—it’ll help us keep the magic flowing smoothly.

> [!IMPORTANT]  
> Before you proceed, please confirm the following:
> - [x] I’ve checked the API docs and confirmed this feature doesn’t already exist (no duplicates, please!).
> - [x] I’ve scoured the existing issues to make sure this hasn’t been requested before.
> - [x] I get that feature requests are reviewed periodically and might take a little time—patience is key!

---

**Describe the Feature Request**

Add a module in which a user can run basic prediction tasks, such as clustering, classification, or regression.

1. With a small configuration effort, the library should run predefined models
2. Tasks like regression should use specific model implementations related to spatial distribution
2.1 For datasets that have only latitude/longitude information, it could use libraries like [pykriging](https://geostat-framework.readthedocs.io/projects/pykrige/en/stable/generated/pykrige.rk.RegressionKriging.html)
2.2 For datasets with geometry (polygons, lines, etc.), it could use libraries like [spreg from PySAL](https://spreg.readthedocs.io/en/latest/generated/spreg.ML_Lag.html)
2.3 For other purposes, it could use scikit-learn, PyTorch, etc.

**Describe Preferred Solution**

**Describe Alternatives**

**Related Code**

There is a new branch [`feat/add_model`](https://github.com/VIDA-NYU/UrbanMapper/tree/feat/add_model)

**Additional Context**

**If the feature request is approved, would you be willing to submit a PR?**

- [x] Yes
- [ ] No

_(No worries if you’re new to this—we’ll guide you every step of the way! 🤝)_

**Comment by simonprovost** [comment]
So so cool feature @fabiofelix ! May I suggest that this is an optional module via `uv add urban-mapper[ml]` or `pip install urban-mapper[ml]`, so that it does not overweight the package if people won't work with ML downstream tasks following a urban mapper pipeline analysis?

Sklearn, Pytorch can be quite heavy :) Much better to be installed only and only if you've got the wish to work with  ML in downstream tasks.

Cheers!

**Comment by fabiofelix** [comment]
Scikit-learn is already a dependency, and PyTorch is only necessary if a user wants to implement a customized model with it.

**Comment by simonprovost** [comment]
@fabiofelix Just changed the title as `model` only was confusing me all the time I am opening the issues tab. Feel free to change to something else for more accuracy if needed! cheers.
